internal/dispatch/caching/caching.go

```diff
@@ -164,8 +165,11 @@ func (cd *CachingDispatcher) DispatchCheck(ctx context.Context, req *v1.Dispatch
 
 	// We only want to cache the result if there was no error
 	if err == nil {
-		toCache := checkResultEntry{computed, computed.Metadata.DepthRequired}
-		toCache.result.Metadata.DispatchCount = 0
+		adjustedComputed := proto.Clone(computed).(*v1.DispatchCheckResponse)
+		adjustedComputed.Metadata.CachedDispatchCount = adjustedComputed.Metadata.DispatchCount
+		adjustedComputed.Metadata.DispatchCount = 0
+
+		toCache := checkResultEntry{adjustedComputed, computed.Metadata.DepthRequired}
```

Comment:
Do we want to just also include the depth required in the response metadata, since there is a limit?

Reply:
Sure, we can use that directly if you like

Reply:
I switched to use the metadata's var, but kept the entries in case there is anything we want to add the cache entries. Let me know if I should just store the response message directly in the cache as a pointer (which might have its own concerns)

Reply:
I think keeping the entry layer of indirection is fine.